src/node_buffer.cc

```diff
@@ -148,11 +148,13 @@ void CallbackInfo::WeakCallback(
 
 
 void CallbackInfo::WeakCallback(Isolate* isolate, Local<Object> object) {
-  SPREAD_ARG(object, obj);
-  CHECK_EQ(obj_offset, 0);
-  CHECK_EQ(obj_c.ByteLength(), obj_length);
+  CHECK(object->IsArrayBuffer());
+  Local<ArrayBuffer> buf = object.As<ArrayBuffer>();
+  ArrayBuffer::Contents obj_c = buf->GetContents();
+  char* const obj_data = static_cast<char*>(obj_c.Data());
+  CHECK_NE(obj_data, nullptr);
```

Comment:
@TooTallNate has mentioned how some use a zero-length buffer to keep track of things in their modules. A zero-length buffer would be `nullptr`. Instead should be like:
````cpp
if (obj_c.Length() > 0) CHECK_NE(obj_data, nullptr);
```